src/components/shapes/draw.js

```diff
@@ -605,13 +607,30 @@ function drawLabel(gd, index, options, shapeGroup) {
     // Remove existing label
     shapeGroup.selectAll('.shape-label').remove();
 
-    // If no label, return
-    if(!options.label.text) return;
+    // If no label text or texttemplate, return
+    if(!(options.label.text || options.label.texttemplate)) return;
+
+    // Text template overrides text
```

Comment:
Let `defaults` do the overriding - ie don't coerce `text` if there's a `texttemplate`.

Reply:
and even better - don't try to coerce `texttemplate` if `type==='path'` - then you don't need the `if(options.type !== 'path')` below :)

Reply:
@alexcjohnson Changed the coercion in `defaults.js` but I believe we still need the `if(texttemplate)` statement (see updated code) -- correct me if wrong!